app/javascript/components/patient/close_contacts/CloseContactTable.js

```diff
@@ -0,0 +1,474 @@
+   * @param {*} newCloseContactData
+   * @param {*} isEdit - whether this is an edit or a new close contact
+   */
+  updateCloseContact = (ccData, isEdit) => {
+    this.setState({ isLoading: true }, () => {
+      let url = `${window.BASE_PATH}/close_contacts`;
+      if (isEdit) {
+        url += `/${ccData.id}`;
+      }
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .post(url, {
+          patient_id: this.props.patient.id,
+          first_name: ccData.first_name || '',
+          last_name: ccData.last_name || '',
+          primary_telephone: ccData.primary_telephone ? formatPhoneNumber(ccData.primary_telephone) : '',
+          email: ccData.email || '',
+          last_date_of_exposure: ccData.last_date_of_exposure || null,
+          assigned_user: ccData.assigned_user || null,
+          notes: ccData.notes || '',
+          enrolled_id: ccData.enrolled_id || null,
+          contact_attempts: ccData.contact_attempts || 0,
+        })
+        .then(() => {
+          location.reload();
+        })
+        .catch(error => {
+          this.setState({ isLoading: false });
+          reportError(error);
+        });
+    });
+  };
+
+  /**
+   * Creates the action button & dropdown for each row in the table.
+   * @param {Object} data - Data about the cell this filter is called on.
+   */
+  renderActionsDropdown = data => {
+    const rowIndex = data.rowIndex;
+    const rowData = data.rowData;
+    // Set the direction to be "up" when there are not enough rows in the table to have space for the dropdown.
+    // The table custom class handles the rest.
+    // NOTE: If this dropdown increases in height, the custom table class passed to CustomTable will need to be updated.
+    const direction = this.state.table.rowData && this.state.table.rowData.length > 4 ? null : 'up';
+    return (
+      <Dropdown drop={direction} className="close-contact-action-button">
+        <Dropdown.Toggle
+          id={`close-contact-action-button-${rowData.id}`}
+          size="sm"
+          variant="primary"
+          aria-label={`Close Contact Action Button for Close Contact ${rowData.id}`}>
+          <i className="fas fa-cogs fw" />
+        </Dropdown.Toggle>
+        <Dropdown.Menu drop={direction}>
+          <Dropdown.Item className="px-4" onClick={() => this.toggleEditModal(rowIndex)}>
+            <i className="fas fa-edit fa-fw" />
+            <span className="ml-2"> Edit</span>
+          </Dropdown.Item>
+          <Dropdown.Item className="px-4" onClick={() => this.handleContactAttempt(rowIndex)}>
+            <i className="fas fa-phone fa-flip-horizontal" />
+            <span className="ml-2">Contact Attempt</span>
+          </Dropdown.Item>
+          {rowData.enrolled_id && (
+            <Dropdown.Item className="px-4" onClick={() => (location.href = `${window.BASE_PATH}/patients/${rowData.enrolled_id}`)}>
```

Comment:
This should include the nav query param. You can use the `patientHref` method. https://github.com/SaraAlert/SaraAlert/blob/1deaafc2a2d9b6b61fc989166dcab401ca78fb78/app/javascript/utils/Navigation.js#L13-L15

This means the `CloseContactTable` component will also need the `workflow` passed in to the props.

Reply:
fixed.